Src/IronPython/Runtime/Binding/PythonProtocol.cs

```diff
@@ -108,7 +108,21 @@ static partial class PythonProtocol {
                 );
             }
 
-            return Ast.NotEqual(callAsInt, AstUtils.Constant(0));
+            var res = Expression.Parameter(typeof(int));
+            return Ast.Block(
+                new[] { res },
+                Ast.Assign(res, callAsInt),
+                Ast.IfThen(Ast.LessThan(res, Ast.Constant(0)),
+                    Ast.Throw(
+                        Ast.Call(
+                            typeof(PythonOps).GetMethod("ValueError"),
```

Comment:
We have to the way this is implemented. It doesn't use the normal __len__ call, so it has to be thrown in here.